galgebra/printer.py

```diff
@@ -801,10 +801,24 @@ def repl_func(m):
 
 
 def tex(paper=(14, 11), debug=False, prog=False, pt='10pt'):
-    """
+    r"""
     Post processes LaTeX output (see comments below), adds preamble and
     postscript.
 
+    This postprocessing includes:
+
+    * ``##`` at the start of a line toggles postprocessing on and off. It
+      begins as on. The rest of the line is never processed.
+    * ``#%`` at the start of a line wraps the line in an ``equation`` environment,
+      unless the line already contains an ``align`` environment, in which case
+      everything before the ``align`` environment is moved inside it.
+    * ``#`` at the start of a line disables postprocessing for that line only
+    * Otherwise, lines are split at ``=``. The left-hand-side is post-processed
+      from python syntax into TeX, unless it starts with a ``%``. Finally,
+      the line is wrapped in an ``equation`` environment,
+      unless the line already contains an ``align`` environment, in which case
+      everything before the ``align`` environment is moved inside it
```

Comment:
Rendering is at least not awful: https://galgebra--379.org.readthedocs.build/en/379/generated/galgebra.printer.html#galgebra.printer.tex

Reply:
Oh this logic in code is messy and reflected in the doc......

Reply:
Absolutely...

Reply:
I would suggest the following structure of the doc:

- Start by stating that the postprocess includes two parts:
  1. texifying lhs
  2. wrapping it with equation env unless already wrapped with align
- Then explain the various markers:
  - # disables postprocessing for that line only
  - % disables the part 1 of postprocessing for that line only
  - #% (seems to be exactly the same as only %?)
  - ## toggles postprocessing, i.e. the first ## turns it off, the second ## turns it back on, and so on (The text after this marker is never postprocessed.)

(Sorry I didn't spell every word out, only sketching the big picture.

Reply:
You're right, `%` and `#%` look to be practically identical...

Reply:
Updated to merge their implementations, the only effect was insignificant whitespace.